source/agora/utils/Test.d

```diff
@@ -589,3 +591,174 @@ unittest
         assert(!verify(pz.V, ssz, "WK.Keys.z".representation));
     }
 }
+
+/*******************************************************************************
+
+    Generate a new transaction that evenly splits the input accross parties
+
+*******************************************************************************/
+
+public Transaction split (TxType type = TxType.Payment)
+    (const ref Transaction input, scope const KeyPair[] from,
+     scope const PublicKey[] toward...)
+    @safe
+{
+    Amount amount;
+    if (!input.getSumOutput(amount))
+        assert(0, "Invalid transaction passed to `split`");
+
+    auto remainder = amount.div(cast(uint) toward.length);
+    Transaction result = Transaction(type);
+    foreach (addr; toward)
+        result.outputs ~= Output(amount, addr);
+
+    // Add the remainder to the first output.
+    result.outputs[0].value.mustAdd(remainder);
+    const inputHash = input.hashFull();
+
+    // Add support for Transactions with multiple recipients
+    foreach (idx, const ref _; input.outputs)
+        result.inputs ~= Input(inputHash, cast(uint) idx);
+
+    auto sign = (KeyPair kp, Hash h) @trusted { return kp.secret.sign(h[]); };
+
+    const resultHash = result.hashFull();
+    foreach (idx, ref in_; result.inputs)
+    {
+        auto rng = from.find!(a => a.address == input.outputs[idx].address);
+        assert(rng.length);
+        const owner = rng[0];
+        in_.signature = sign(owner, resultHash);
+    }
+    return result;
+}
+
+/// Test for a split with the same amount of outputs as inputs
+/// Essentially doing an equality transformation
+unittest
+{
+    import std.range;
+
+    KeyPair[] keys = iota(8).map!(_ => KeyPair.random()).array;
+    KeyPair genesisKP = getGenesisKeyPair();
+    const first = GenesisBlock.txs[0];
+    const equalTx = first.split([genesisKP], keys.map!(k => k.address).array);
+    // This transaction has 8 txs, hence it's just equality
+    assert(equalTx.inputs.length == 8);
+    assert(equalTx.outputs.length == 8);
+    // Since the amount is evenly distributed in Genesis,
+    // they all have the same value
+    const ExpectedAmount = first.outputs[0].value;
+    assert(equalTx.outputs.all!(val => val.value == ExpectedAmount));
+}
+
+/// Test with twice as many outputs as inputs
+unittest
+{
+    import std.range;
+
+    KeyPair[] keys = iota(16).map!(_ => KeyPair.random()).array;
+    // Use Genesis
+    KeyPair genesisKP = getGenesisKeyPair();
+    const first = GenesisBlock.txs[0];
+    const equalTx = first.split([genesisKP], keys.map!(k => k.address).array);
```

Comment:
```suggestion
    const result = first.split([genesisKP], keys.map!(k => k.address).array);
```

I used `equalTx` because it was an "equality" transaction, but here it doesn't make sense.
For consistency, you can change all `equalTx` to `result` (even the first one).